src/Statement.php

```diff
@@ -13,88 +13,41 @@
 class Statement extends \Doctrine\DBAL\Statement
 {
     /**
-     * The connection this statement is bound to and executed on.
-     *
-     * @var Connection
+     * Recreates the statement for retry.
      */
-    protected $conn;
-
-    /**
-     * @var mixed[][]
-     */
-    private $boundValues = [];
-
-    /**
-     * @var mixed[][]
-     */
-    private $boundParams = [];
+    private function recreateStatement(): void
+    {
+        $this->stmt = $this->conn->getWrappedConnection()->prepare($this->sql);
+    }
 
-    /**
-     * @var mixed[]|null
-     */
-    private $fetchMode;
+//    public function execute($params = null): Result
+//    {
+//        return $this->executeWithRetry([parent::class, 'execute'], $params);
+//    }
```

Comment:
I'm leaving this here because we will need to re-add this method lather. I will probably try to split it into a child class, so that it could be enabled when needed, because it will trigger deprecations.